main.go

```diff
@@ -39,46 +39,51 @@ func main() {
 		}
 	}
 
-	content, readAllErr := ioutil.ReadAll(os.Stdin)
-	if readAllErr != nil {
-		fmt.Fprintln(os.Stderr, readAllErr)
-		os.Exit(1)
-	}
-
-	if len(content) <= 0 {
-		return
-	}
-
+	buf := bufio.NewReader(os.Stdin)
 	var data, parameterBasedData [][][]byte
 	var lineCounter int
 	delimiter := []byte{9}
 
-	contentLineSplit := bytes.Split(content, []byte{'\n'})
-	data = make([][][]byte, len(contentLineSplit))
+	for {
+		content, readAllErr := buf.ReadBytes('\n')
+		if len(content) < 1 {
```

Comment:
What's this for?

Reply:
You said i should make it exactly like you did.

Reply:
Right, but what is this `if` for?

Reply:
without the if im getting an end of file error. the if avoids it

Reply:
... along with all other errors. Please check explicitly for `readAllErr == EOF` inside the next if instead.